fasta2a/fasta2a/applications.py

```diff
@@ -0,0 +1,124 @@
+from __future__ import annotations as _annotations
+
+from collections.abc import AsyncIterator, Sequence
+from contextlib import asynccontextmanager
+from typing import Any
+
+from starlette.applications import Starlette
+from starlette.middleware import Middleware
+from starlette.requests import Request
+from starlette.responses import Response
+from starlette.routing import Route
+from starlette.types import ExceptionHandler, Lifespan, Receive, Scope, Send
+
+from .schema import AgentCard, Provider, Skill, a2a_request_ta, a2a_response_ta, agent_card_ta
+from .storage import Storage
+from .task_manager import TaskManager
+from .worker import Worker
+
+
+class FastA2A(Starlette):
+    """The main class for the FastA2A library."""
+
+    def __init__(
+        self,
+        *,
+        storage: Storage,
+        worker: Worker,
+        # Agent card
+        name: str | None = None,
+        url: str = 'http://localhost:8000',
+        version: str = '1.0.0',
+        description: str | None = None,
+        provider: Provider | None = None,
+        skills: list[Skill] | None = None,
+        # Starlette
+        debug: bool = False,
+        routes: Sequence[Route] | None = None,
+        middleware: Sequence[Middleware] | None = None,
+        exception_handlers: dict[Any, ExceptionHandler] | None = None,
+        lifespan: Lifespan[FastA2A] | None = None,
+    ):
+        if lifespan is None:
+            lifespan = _default_lifespan
+
+        super().__init__(
+            debug=debug,
+            routes=routes,
+            middleware=middleware,
+            exception_handlers=exception_handlers,
+            lifespan=lifespan,
+        )
+
+        self.name = name or 'Agent'
+        self.url = url
+        self.version = version
+        self.description = description
+        self.provider = provider
+        self.skills = skills or []
+        # NOTE: For now, I don't think there's any reason to support any other input/output modes.
```

Comment:
Would other input/output modes be necessary for multi-modal inputs/outputs? If so, you might want to not emphasize that as a value prop in the docs (it currently lists "- **Rich Communication**: Support for multimodal content exchange (text, structured data, files)" as something the protocol "facilitates").